### PR TITLE
Adds a test for posix_memalign in stdlib.c

### PR DESCRIPTION
**Pull request by mmuesly**
This commit tests the changes introduced in
15aa61f4acb865addf287f340dc0e3d1a3b0dbe6

### CODE REVIEW

**Review by tautschnig — APPROVED** [review]

File: regression/cbmc/posix_memalign/main.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#include <malloc.h>
```
Comment (tautschnig) [inline regression/cbmc/posix_memalign/main.c]: This header isn't portable (as CI reminded us), but also not necessary. Please just remove that line.

**Review by tautschnig — CHANGES_REQUESTED** [review]

**Comment by mmuesly** [comment]
Travis output:
```The command "make -C jbmc/src "CXX=${COMPILER} ${EXTRA_CXXFLAGS}" -j3" failed and exited with 2 during``` 

The package jbmc/src isn't touched in this commit, so I don't get, why this don't pass travis checks. The build error seems to be related to some Java classes.

Could you provide some guidance regarding a solution of the broken travis check?

**Comment by kroening** [comment]
PR #2654 should fix Travis!

**Review by kroening — APPROVED** [review]

File: regression/cbmc/posix_memalign/main.c (hunk 1 above)
Comment (mmuesly) [reply regression/cbmc/posix_memalign/main.c]: I addressed the requested changes in 3a11d7d

**Comment by tautschnig** [comment]
@mmuesly Thanks a lot for the fix. Could you please rebase it on the latest version of develop to make CI happy?

**Review by tautschnig — APPROVED** [review]

**Comment by mmuesly** [comment]
Rebased. But for some reason, the cbmc-windows CI is unhappy now. This test had passed before the rebase. Might you share some insights, why this test fails now?

**Comment by tautschnig** [comment]
@peterschrammel @kroening Could you please re-start the Windows build? It failed with
```
make: Entering directory '/cygdrive/c/codebuild/tmp/src937370995/src/github.com/diffblue/cbmc/src'
Downloading Minisat 2.2.1
--2018-08-02 18:00:01--  http://ftp.debian.org/debian/pool/main/m/minisat2/minisat2_2.2.1.orig.tar.gz
Resolving ftp.debian.org (ftp.debian.org)... 130.89.148.12, 2001:67c:2564:a119::148:12
Connecting to ftp.debian.org (ftp.debian.org)|130.89.148.12|:80... failed: Connection timed out.
Connecting to ftp.debian.org (ftp.debian.org)|2001:67c:2564:a119::148:12|:80... failed: Network is unreachable.
make: *** [Makefile:94: minisat2-download] Error 4
```

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 00b26a8fdeb19295f4e79f8afd2634eb479c0d5d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/80792537
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).
##

**Comment by kroening** [comment]
Restarted. Note that CodeBuild won't update the badge.

**Comment by mmuesly** [comment]
Thanks!

**Pull request merged**